src/retest/main.c

```diff
@@ -465,6 +465,83 @@ error_record_init(struct error_record *erec)
 	return 0;
 }
 
+static char
+nibble_char(unsigned hex) {
+	static const char *dig = "0123456789abcdef";
+
+	return dig[hex & 0xf];
+}
+
+static char *
+dup_str_esc(const char *s, int *err)
+{
+	size_t len, numesc;
+	const char *sp;
+	char *cpy, *cp;
+
+	if (err != NULL && *err != 0) {
+		return NULL;
+	}
+
+	len = 0;
+	numesc = 0;
+	for (sp = s; *sp; sp++) {
+		if (*sp != ' ' && isspace((unsigned char)(*sp))) {
+			len += 2;
+			numesc++;
+		} else if (!isprint((unsigned char)*sp)) {
+			len += 4;
+			numesc++;
+		} else {
+			len++;
+		}
+	}
+
+	cpy = malloc(len+1);
```

Comment:
`xmalloc()` would make sense here, we don't care to handle the error specifically for a CLI tool